solarkraft/src/fetcher/callDecoder.ts

```diff
@@ -58,6 +59,15 @@ export async function extractContractCall(
     const method = params[1]
     const methodArgs = params.slice(2)
 
+    // Now we look into the typemap file, to see if we're given type hints. This is effectively required
+    // for vector-like arguments, which could be encodings of enums
+    // Could be undefined, in which case we should fail on ambiguous input
+    const typeHints = {
+        variables: typemapJson['variables'],
+    }
+    typeHints[method] =
+        'methods' in typemapJson ? typemapJson['methods'][method] : undefined
```

Comment:
I guess you could, I was just thinking that you could prune out method annotations for methods not referenced in this tx, but maybe that's redundant.
> Also, there's the rare case where this would clash if a method is named variables 😄

Actually true, and probably a good reason not to do what I did here.